ovh/resource_cloud_project_network_private_subnet_v2.go

```diff
@@ -0,0 +1,267 @@
+	networkId := d.Get("network_id").(string)
+	subnetName := d.Get("name").(string)
+	cidr := d.Get("cidr").(string)
+	enableGatewayIP := d.Get("enable_gateway_ip").(bool)
+	enableDHCP := d.Get("dhcp").(bool)
+	gatewayIp := d.Get("gateway_ip").(string)
+
+	hostRoutesStrings, err := helpers.StringMapFromSchema(d, "host_routes", "destination", "nexthop")
+	if err != nil {
+		return err
+	}
+
+	hostRoutes := []HostRoute{}
+	for _, hostRouteStrings := range hostRoutesStrings {
+		hostRoutes = append(hostRoutes, HostRoute{
+			Destination: hostRouteStrings["destination"],
+			Nexthop:     hostRouteStrings["nexthop"],
+		})
+	}
+
+	allocationPoolsStrings, err := helpers.StringMapFromSchema(d, "allocation_pools", "start", "end")
+	if err != nil {
+		return err
+	}
+
+	allocationPools := []AllocationPool{}
+	for _, allocationPoolStrings := range allocationPoolsStrings {
+		allocationPools = append(allocationPools, AllocationPool{
+			Start: allocationPoolStrings["start"],
+			End:   allocationPoolStrings["end"],
+		})
+	}
+
+	dnsNameServers, err := helpers.StringsFromSchema(d, "dns_nameservers")
+	if err != nil {
+		return err
+	}
+
+	createSubnetParams := &CloudProjectNetworkPrivateV2CreateOpts{
+		Name:            subnetName,
+		Cidr:            cidr,
+		IpVersion:       4,
+		AllocationPools: allocationPools,
+		DnsNameServers:  dnsNameServers,
+		GatewayIp:       gatewayIp,
+		EnableGatewayIP: enableGatewayIP,
+		EnableDHCP:      enableDHCP,
+		HostRoutes:      hostRoutes,
+	}
+
+	subnetResponse := &CloudProjectNetworkPrivateV2Response{}
+	endpointPostSubnet := fmt.Sprintf("/cloud/project/%s/region/%s/network/%s/subnet",
+		url.PathEscape(serviceName),
+		url.PathEscape(regionName),
+		url.PathEscape(networkId),
+	)
+	err = config.OVHClient.Post(endpointPostSubnet, createSubnetParams, subnetResponse)
+	if err != nil {
+		return fmt.Errorf("calling POST %s with params %v:\n\t %q", endpointPostSubnet, createSubnetParams, err)
+	}
+
+	d.Set("gateway_ip", subnetResponse.GatewayIp)
+	d.Set("cidr", subnetResponse.Cidr)
+	d.Set("enable_gateway_ip", subnetResponse.GatewayIp != nil)
+	d.Set("service_name", serviceName)
+	d.Set("network_id", networkId)
+	d.Set("dhcp", subnetResponse.DHCPEnabled)
+	d.Set("region", regionName)
+	d.SetId(subnetResponse.Id)
+	log.Printf("[DEBUG] Read Public Cloud Private Network %v", subnetResponse)
+	return nil
+}
+
+func resourceCloudProjectNetworkPrivateSubnetV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	networkId := d.Get("network_id").(string)
+	regionName := d.Get("region").(string)
+
+	subnets := []*CloudProjectNetworkPrivateV2Response{}
+
+	log.Printf("[DEBUG] Will read public cloud private network subnet for project: %s, region: %s, network: %s, id: %s", serviceName, regionName, networkId, d.Id())
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/region/%s/network/%s/subnet", serviceName, regionName, networkId)
```

Comment:
why don't you just fetch the right subnet using `GET /cloud/project/{serviceName}/region/{regionName}/network/{networkId}/subnet/{subnetId}` ?